spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -366,6 +366,14 @@ class CometSparkSessionExtensions
         }
       }
 
+      /**
+       * Convert operator to proto and then apply a transformation to wrap the proto in a new
+       * plan.
+       */
+      def wrapProto(op: SparkPlan, fun: Operator => SparkPlan): SparkPlan = {
```

Comment:
Can we call this something like `newPlanWithProto`? I'm fine with `wrapProto` as well.

Reply:
Updated